external-crates/move/crates/move-analyzer/src/analysis/typing_analysis.rs

```diff
@@ -676,24 +679,6 @@ impl TypingAnalysisContext<'_> {
 
         if let Some(exp) = &arm.guard {
             self.visit_exp(exp);
-            // Enum guard variables have different type (immutable reference) than variables in
```

Comment:
This appeared to be entirely redundant - the only thing we need (which is guard location) is already encode in `DefInfo` so this code could have been removed independently